### PR TITLE
fix: Install polyfills for Comcast X1 devices

### PR DESCRIPTION
**Pull request by Iragne**
Close https://github.com/shaka-project/shaka-player/issues/7522

### CODE REVIEW

**Comment by Iragne** [comment]
Still testing it

File: lib/util/platform.js [hunk 1]
```diff
@@ -354,6 +355,15 @@ shaka.util.Platform = class {
     return shaka.util.Platform.userAgentContains_('Sky_STB');
   }
 
+  /**
+   * Check if the current platform is XOne.
+   */
+  static isXOne() {
+    return navigator.platform &&
+        navigator.platform.includes('Linux aarch64') &&
+        shaka.util.Platform.userAgentContains_('AppleWebKit');
```
Comment (joeyparrish) [inline lib/util/platform.js:364]: Surely there is some more specific attribute you can check.  Maybe something outside the UA string, if the UA looks very generic?  I have to imagine any WebKit build on Arm Linux would be detected as XOne with this check.

Comment (Iragne) [reply lib/util/platform.js:364]: Agree but today i checked 


    navigator.platform
    < "Linux aarch64"

    navigator.userAgent
    < "Mozilla/5.0 (Linux; x86_64 GNU/Linux) AppleWebKit/601.1 (KHTML, like Gecko) Version/8.0 Safari/601.1 WPE"

    navigator.vendor
    < "Apple Computer, Inc."

Any other idea ?

Comment (avelad) [reply lib/util/platform.js:364]: Maybe you can include `Version/8.0`

Comment (Iragne) [reply lib/util/platform.js:364]: ok

Comment (joeyparrish) [reply lib/util/platform.js:364]: Does this device expose any unusual properties anywhere within the global namespace?  Any API specific to this device?

Comment (Iragne) [reply lib/util/platform.js:364]: Let me check i will stringify all ;)

Comment (Iragne) [reply lib/util/platform.js]: I found that we have the RDK as a global variable. I used it too

File: lib/util/platform.js [hunk 2]
```diff
@@ -354,6 +355,23 @@ shaka.util.Platform = class {
     return shaka.util.Platform.userAgentContains_('Sky_STB');
   }
 
+  /**
+   * Check if the current platform is XOne.
+   */
+  static isXOne() {
+    let rdk = false;
+    if (window && Object.keys(window).includes('ServiceManager')) {
```
Comment (joeyparrish) [inline lib/util/platform.js:363]: Does this simpler version work instead?

```js
01234567890123456789012345678901234567890123456789012345678901234567890123456789
if (window.ServiceManager && window.ServiceManager.getServiceForJavaScript &&
    navigator.platform.includes('Linux aarch64') &&
    shaka.util.Platform.userAgentContains_('AppleWebKit') &&
    shaka.util.Platform.userAgentContains_('Version/8.0')) {
```

You will need to add a file like `externs/xbox.js` to satisfy the compiler that `ServiceManager` and `getServiceForJavaScript` might exist.  The types can be very simple if you don't need to call into any of that functionality.

Comment (Iragne) [reply lib/util/platform.js:363]: Thanks i did it and also refactor the previous check

File: lib/util/platform.js [hunk 3]
```diff
@@ -354,6 +354,18 @@ shaka.util.Platform = class {
     return shaka.util.Platform.userAgentContains_('Sky_STB');
   }
 
+  /**
+   * Check if the current platform is Comcast X1.
+   */
+  static isComecastX1() {
```
Comment (avelad) [inline lib/util/platform.js:360]: `isComecastX1` or `isComcastX1`?

Comment (Iragne) [reply lib/util/platform.js:360]: Sorry

**Review by avelad — APPROVED** [review]

File: externs/comcast.js
Comment (tykus160) [inline externs/comcast.js:1]: Do we really need this extern? Can't we detect X1 without looking into some API existence that we don't use anyway?

Comment (Iragne) [reply externs/comcast.js:1]: Hi @tykus160  

here it was what i shared we can use WPE but i was quite afraid of using it
Happy to reshape to remove the extern 
Here the detail 
navigator.platform
< "Linux aarch64"

navigator.userAgent
< "Mozilla/5.0 (Linux; x86_64 GNU/Linux) AppleWebKit/601.1 (KHTML, like Gecko) Version/8.0 Safari/601.1 WPE"

navigator.vendor
< "Apple Computer, Inc."

Comment (tykus160) [reply externs/comcast.js:1]: Now I see we already have a method for detecting X1 in Platform utils: https://github.com/shaka-project/shaka-player/blob/main/lib/util/platform.js#L367C1-L373C4

Comment (Iragne) [reply externs/comcast.js:1]: Yes to remove it and no patch associated to it

Comment (tykus160) [reply externs/comcast.js:1]: So you need to add WPE check to polyfills, that would be it :)

Comment (Iragne) [reply externs/comcast.js:1]: Yep

Comment (Iragne) [reply externs/comcast.js:1]: done

**Review by tykus160 — APPROVED** [review]

**Pull request merged**